### PR TITLE
:recycle: Remove redis select

### PR DESCRIPTION
**Pull request by rinx**

### CODE REVIEW

File: pkg/meta/redis/service/redis.go [hunk 1]
```diff
@@ -163,57 +169,57 @@ func (c *client) Connect(ctx context.Context) error {
...
 	}
-	vals = make([]string, 0, len(keys))
-	for _, r := range res.Val() {
-		val, ok := r.(string)
-		if ok {
-			vals = append(vals, val)
+	vals = make([]string, 0, len(ress))
+	for _, res := range ress {
+		if err = res.Err(); err != nil {
+			continue
```
Comment (rinx) [inline pkg/meta/redis/service/redis.go]: Should I use `return nil, err` here?

Comment (rinx) [reply pkg/meta/redis/service/redis.go]: Or revise the interface of this function to return `(vals []string, errs []error)`.

File: pkg/meta/redis/service/redis.go [hunk 2]
```diff
@@ -289,28 +289,30 @@ func (c *client) delete(idx int, key string) (val string, err error) {
...
+	for _, v := range vals {
+		ress = append(ress, pipe.Del(appendPrefix(pfxInv, v)))
 	}
-	if err = v.Err(); err != nil {
+	if _, err = pipe.Exec(); err != nil {
 		return nil, err
 	}
+	for _, res := range ress {
+		if err = res.Err(); err != nil {
+			continue
```
Comment (rinx) [inline pkg/meta/redis/service/redis.go]: Same as above.

File: pkg/meta/redis/service/redis.go [hunk 3]
```diff
@@ -163,57 +174,59 @@ func (c *client) Connect(ctx context.Context) error {
...
-	vals = make([]string, 0, len(keys))
-	for _, r := range res.Val() {
-		val, ok := r.(string)
-		if ok {
-			vals = append(vals, val)
+	var errs error
+	vals = make([]string, 0, len(ress))
+	for _, res := range ress {
+		if err = res.Err(); err != nil {
+			errs = errors.Wrap(errs, err.Error())
```
Comment (kpango) [inline pkg/meta/redis/service/redis.go]: could you add more information ? we need to know which key make error and if you implement it you shoud define error function in the internal/errors package

File: pkg/meta/redis/service/redis.go [hunk 4]
```diff
@@ -223,60 +236,49 @@ func (c *client) Set(key, val string) error {
...
-	vk := pipe.MSet(vkl...)
-	if _, err := pipe.Exec(); err != nil {
+	if _, err = pipe.Exec(); err != nil {
 		return err
 	}
-	if err := kv.Err(); err != nil {
-		return err
+	var errs error
+	for _, res := range ress {
+		errs = errors.Wrap(errs, res.Err().Error())
```
Comment (kpango) [inline pkg/meta/redis/service/redis.go]: could you please check res.Err is nil or not here, also errors.Wrap checks error and message, but it is depends on library

File: pkg/meta/redis/service/redis.go (hunk 3 above)
Comment (rinx) [reply pkg/meta/redis/service/redis.go]: Okay. I'll try it. Thanks! :bow:

File: pkg/meta/redis/service/redis.go [hunk 5]
```diff
@@ -223,60 +236,51 @@ func (c *client) Set(key, val string) error {
...
-	if _, err := pipe.Exec(); err != nil {
+	if _, err = pipe.Exec(); err != nil {
 		return err
 	}
-	if err := kv.Err(); err != nil {
-		return err
+	var errs error
+	for _, res := range ress {
+		if err = res.Err(); err != nil {
+			errs = errors.Wrap(errs, err.Error())
```
Comment (kpango) [inline pkg/meta/redis/service/redis.go]: same as above

File: internal/errors/errors.go [hunk 6]
```diff
@@ -30,8 +30,20 @@ import (
 var (
 
 	// Redis
-	ErrRedisInvalidKVVKIndex = func(kv, vk int) error {
-		return Errorf("kv index and vk index must be defferent.\t(kv: %d,\tvk: %d)", kv, vk)
+	ErrRedisInvalidKVVKPrefix = func(kv, vk string) error {
+		return Errorf("kv index and vk prefix must be defferent.\t(kv: %s,\tvk: %s)", kv, vk)
+	}
+
+	ErrRedisFetchingValueFailed = func(key string, msg string) error {
```
Comment (kpango) [inline internal/errors/errors.go]: msgはerror型で受け取る方が綺麗かなーと思ったり！

Comment (kpango) [reply internal/errors/errors.go]: ErrRedisGetOperationFailed かな

**Comment by kpango** [comment]
LGTM

**Pull request merged**